lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -103,6 +91,14 @@ sub needle_url {
 # Call to viewimg or viewaudio
 sub view {
     my ($self) = @_;
+
+    # Redirect users with the old preview link
+    if (!$self->req->is_xhr) {
+        my $anchor = "#step/" . $self->param('moduleid') . "/" . $self->param('stepid');
+        my $target_url = $self->url_for('test', testid => $self->param('testid'));
+        return $self->redirect_to($target_url . $anchor);
+    }
```

Comment:
where is the test for this? you rewrote the test to rely on the new way only, but can you preserve the deprecated way in tests, too?